### PR TITLE
Add table row hover and remove highlighted column styling

### PR DESCRIPTION
**Pull request by leemunroe**
Proposal to improve table styling.

1. Add a row hover style. Why?  Improves usability by visually highlighting the row the user is looking at so they can see the related data and columns.

2. Remove the highlighted active column. Why? Don't think this is adding value (as the caret already shows the active column) and it looks odd whenever there are just two columns.

3. Show the sorting caret on hover. Why? So the user knows they can interact with and sort the table.

Before
https://cl.ly/3Z40263T2c1y

After
https://cl.ly/0U05022x2M3p

cc @ashenden

### CODE REVIEW

**Comment by ashenden** [comment]
Fully support these changes.

File: src/styles/content/tables/styles.less [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
+    }
+  }
+
   tr,
   .table-row {
 
     &:hover {
 
+      td,
+      td.active {
```
Comment (jfurrow) [inline src/styles/content/tables/styles.less]: Sorry for the nit-pick but let's write this is as follows:

```LESS
td {

  &,
  &.active {
    background-color: @table-cell-hover-color;
  }
}
```

Comment (leemunroe) [inline src/styles/content/tables/styles.less]: Sounds good. Done.

**Comment by ashenden** [comment]
@leemunroe @jfurrow you should be able to achieve the row hover behavior without modifying the CSS.  CNVS provides a class for this called `.table-hover` (http://mesosphere.github.io/cnvs/content/tables/)

**Comment by leemunroe** [comment]
That would mean adding another class to every instance of a table (~10 JS files) and developers having to remember to do that in the future for new tables VS changing one CSS file. Seems like bad practice, no? Isn't the ideal world a reusable semantic class like  `table-collection` or `table-data` instead of adding more classes?

**Comment by jfurrow** [comment]
I agree with @leemunroe that this should be added by default to all tables.

**Comment by ashenden** [comment]
@jfurrow @leemunroe agree, but aren't developers already today using a generalized table component?  One where they don't have to think about classes?  If so, on this generalized table component, could we not define `table-hover` once and then it is just applied to all instances of that table going forward?

What I'm suggesting we avoid is having to manually change the style definition, essentially adding more style, when this is already support and available to us.

**Comment by jfurrow** [comment]
The generalized table component is in `reactjs-components` and we shouldn't add the class there IMO.

**Comment by ashenden** [comment]
@jfurrow think we could talk about this in person tomorrow? I believe we want the same thing (maybe?).

**Comment by jfurrow** [comment]
For sure! Grab me when you're available

**Comment by MatApple** [comment]
@jfurrow @leemunroe what was the outcome of your offline discussion?

**Comment by leemunroe** [comment]
Added `table-hover` class to all relevant tables as per offline discussion.

**Comment by jfurrow** [comment]
@MatApple We decided, for now, to just add the `table-hover` classes to all `Table` components with interactive elements.

**Comment by jfurrow** [comment]
Thanks @leemunroe! Works great.

**Pull request merged**